### ISSUE TITLE
Import statement in declaration file causes silent failure (declaration is completely ignored)

### ISSUE THREAD

**Issue by JakobJingleheimer** [microsoft/TypeScript#55019]
# Bug Report

### 🔎 Search Terms

* declaration file
* `d.ts`
* ESM import / ES module import
* silent failure

### 🕗 Version & Regression Information

Typescript: `5.1.3` (and also `5.1.6`)
Node: `18.12.0`

Seemingly related (but appears to be so outdated it's no-longer applicable): #5506

### ⏯ Playground Link

AFAIK, this cannot be reproduced in the playground

### 💻 Code

`global.d.ts`:

```ts
import type { FunctionComponent, SVGAttributes } from 'react'; // ← this (silently) breaks the declaration file

declare module '*.svg' {
  const content: FunctionComponent<SVGAttributes<SVGElement>>;
  // const content: string; // toggle this line and the line above + the import, and the declaration file is okay

  export default content;
}

declare type Int = number;
```

`test.tsx`:

```tsx
import Icon from './test.svg'; // Cannot find module './test.svg' or its corresponding type declaration. ts(2307)

let foo: Int; // Cannot find name 'Int'. ts(2304) type Int = /*unresolved*/ any
```

### 🙁 Actual behavior

Importing (a type) causes the declaration file to silently break (the entire contents of the file is ignored). There are no errors reported in the declaration file itself.

### 🙂 Expected behavior

At the very least, it should not fail silently. But I also expect the import to work.

**Comment by nmain** [comment]
This is still a duplicate of #5506.  The explanation and description given in that ticket are accurate and correct.

**Comment by JakobJingleheimer** [comment]
When I follow the instructions in that issue, I get specific errors from TypeScript telling me specifically to not do those things.

Could you please provide a little more info about the solution? The original answer provides a lot of info about why it's happening but not about the fix.

**Comment by JakobJingleheimer** [comment]
To clarify, the presence of the `import` at the top is what is breaking the declaration file, and if I try to use `/// <reference />`, typescript specifically tells me to use an import

**Comment by nmain** [comment]
(I wish the playground had an easy way to set up multiple-file examples...)

For exactly the canned scenario presented here, use `import(...)` in the type position to prevent the file from being autodetected as a module:

```typescript
declare module '*.svg' {
  const content: import('react').FunctionComponent<import('react').SVGAttributes<SVGElement>>;

  export default content;
}

declare type Int = number;
```

For larger real world scenarios, `declare global { ... }` can be a useful way to create ambient types in modules.

**Comment by JakobJingleheimer** [comment]
Huzzah! That `import()` did the trick, thanks!

I see this was marked "working as intended", which I think is partially accurate; however, the silent failure does not seem like it would be intended.

**Comment by RyanCavanaugh** [comment]
What you wrote is indistinguishable from someone writing that file and *wanting* it to work that way. That's not a "silent failure".
